tests/unit/record_encoder/test_hybrid_record_encoder.py

```diff
@@ -0,0 +1,92 @@
+from unittest.mock import patch
+
+import pytest
+
+from canopy.knowledge_base.models import KBDocChunk
+from canopy.knowledge_base.record_encoder import HybridRecordEncoder
+from canopy.models.data_models import Query
+from tests.unit.stubs.stub_dense_encoder import StubDenseEncoder
+
+
+@pytest.fixture
+def documents():
+    return [KBDocChunk(id=f"doc_1_{i}",
+                       text=f"Sample document {i}",
+                       document_id=f"doc_{i}",
+                       metadata={"test": i},
+                       source="doc_1")
+            for i in range(5)]
+
+
+@pytest.fixture
+def queries():
+    return [Query(text="Sample query 1"),
+            Query(text="Sample query 2"),
+            Query(text="Sample query 3")]
+
+
+@pytest.fixture
+def inner_dimension():
+    return 4
+
+
+@pytest.fixture
+def inner_encoder(inner_dimension):
+    return StubDenseEncoder(dimension=inner_dimension)
+
+
+@pytest.fixture
+def encoder(inner_encoder):
+    return HybridRecordEncoder(inner_encoder, batch_size=2)
+
+
+def test_dimension(encoder, inner_dimension):
+    assert encoder.dimension == inner_dimension
+
+
+def test_init_encoder_invalid_alpha(inner_encoder):
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(inner_encoder, alpha=-1)
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(inner_encoder, alpha=2)
+
+
+def test_encode_documents(encoder, documents, queries):
+    encoded_documents = encoder.encode_documents(documents)
+    for encoded_document in encoded_documents:
+        assert len(encoded_document.values) == encoder.dimension
+        assert len(encoded_document.sparse_values) > 0
+
+
+def test_encode_queries(encoder, queries):
+    encoded_queries = encoder.encode_queries(queries)
+    assert len(encoded_queries) == len(queries)
```

Comment:
I suggest to also here test that there are some sparse values and that they are in the right format

Reply:
Done